extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/SignalFxRegistryProcessor.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.micrometer.deployment.export;
+
+import java.util.function.BooleanSupplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.micrometer.deployment.MicrometerRegistryProviderBuildItem;
+import io.quarkus.micrometer.runtime.MicrometerRecorder;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+import io.quarkus.micrometer.runtime.export.SignalFxMeterRegistryProvider;
+
+/**
+ * Add support for the SignalFx Meter Registry. Note that the registry may not
+ * be available at deployment time for some projects: Avoid direct class
+ * references.
+ */
+public class SignalFxRegistryProcessor {
+    private static final Logger log = Logger.getLogger(SignalFxRegistryProcessor.class);
+
+    static final String REGISTRY_CLASS_NAME = "io.micrometer.signalfx.SignalFxMeterRegistry";
+    static final Class<?> REGISTRY_CLASS = MicrometerRecorder.getClassForName(REGISTRY_CLASS_NAME);
+
+    public static class SignalFxRegistryEnabled implements BooleanSupplier {
+        MicrometerConfig mConfig;
+
+        @Override
+        public boolean getAsBoolean() {
+            boolean test = mConfig.checkRegistryEnabledWithDefault(mConfig.export.signalfx);
+            return REGISTRY_CLASS != null && mConfig.checkRegistryEnabledWithDefault(mConfig.export.signalfx);
```

Comment:
Should the config be captured in a boolean or just used directly?

Reply:
woops!! good catch!
that was a debug.. because I'd done something else stupid. Thanks!